ambry-frontend/src/main/java/com/github/ambry/frontend/TtlUpdateHandler.java

```diff
@@ -143,17 +158,34 @@ private Callback<String> idConverterCallback() {
     private Callback<Void> securityPostProcessRequestCallback(BlobId blobId) {
       return buildCallback(metrics.updateBlobTtlSecurityPostProcessRequestMetrics, result -> {
         String serviceId = RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.SERVICE_ID, true);
-        router.updateBlobTtl(blobId.getID(), serviceId, Utils.Infinite_Time, routerCallback(),
+        router.updateBlobTtl(blobId.getID(), serviceId, Utils.Infinite_Time, routerCallback(blobId),
             QuotaUtils.buildQuotaChargeCallback(restRequest, quotaManager, false));
       }, restRequest.getUri(), LOGGER, finalCallback);
     }
 
     /**
      * After {@link Router#updateBlobTtl} finishes, call {@link SecurityService#processResponse}.
+     * @param blobId the {@link BlobId} to update
      * @return a {@link Callback} to be used with {@link Router#updateBlobTtl}.
      */
-    private Callback<Void> routerCallback() {
+    private Callback<Void> routerCallback(BlobId blobId) {
```

Comment:
can you rename this parameter to convertedBlobId, since there is already a blobIdStr in the class?

Reply:
Sure, updated.